pkg/query/querier.go

```diff
@@ -28,21 +29,60 @@ import (
 	"github.com/thanos-io/thanos/pkg/tracing"
 )
 
+type SeriesStatsReporter func(seriesStats storepb.SeriesStatsCounter)
+
+var NoopSeriesStatsReporter SeriesStatsReporter = func(_ storepb.SeriesStatsCounter) {}
+
+func NewAggregateStatsReporter(stats *[]storepb.SeriesStatsCounter) SeriesStatsReporter {
+	var mutex sync.Mutex
+	return func(s storepb.SeriesStatsCounter) {
+		mutex.Lock()
+		defer mutex.Unlock()
+		*stats = append(*stats, s)
+	}
+}
+
 // QueryableCreator returns implementation of promql.Queryable that fetches data from the proxy store API endpoints.
 // If deduplication is enabled, all data retrieved from it will be deduplicated along all replicaLabels by default.
 // When the replicaLabels argument is not empty it overwrites the global replicaLabels flag. This allows specifying
 // replicaLabels at query time.
 // maxResolutionMillis controls downsampling resolution that is allowed (specified in milliseconds).
 // partialResponse controls `partialResponseDisabled` option of StoreAPI and partial response behavior of proxy.
-type QueryableCreator func(deduplicate bool, replicaLabels []string, storeDebugMatchers [][]*labels.Matcher, maxResolutionMillis int64, partialResponse, enableQueryPushdown, skipChunks bool, shardInfo *storepb.ShardInfo) storage.Queryable
+type QueryableCreator func(
```

Comment:
does `QueryableCreator` needs to be exported as well?

Reply:
Yes yes, this was already exported because it used in other packages.